Improve onboarding, allow no card processing

AlexP11223 opened the pull request:
Implements #443 

![2022-01-21_07-42](https://user-images.githubusercontent.com/5680466/150489084-73979a8b-a200-4174-a5bf-53e554e349ff.png)

![Peek 2022-01-21 07-41](https://user-images.githubusercontent.com/5680466/150489063-d0deaac2-87e7-4574-810a-389d9575f5aa.gif)

AlexP11223 committed:
Remove unused

AlexP11223 committed:
Allow settings items without titles

AlexP11223 committed:
Use different style for sandbox button

AlexP11223 committed:
Allow to specify products value in partner referrals data

AlexP11223 committed:
New onboarding options, buttons

AlexP11223 committed:
Add onboarding header

AlexP11223 committed:
Prevent unsaved data dialog for onboarding checkboxes

AlexP11223 committed:
Update manual credentials toggle

AlexP11223 committed:
Fix img alt

Dinamiko committed:
Merge branch 'trunk' into pcp-370-onboarding

AlexP11223 committed:
Merge branch 'trunk' into pcp-370-onboarding

AlexP11223 committed:
Merge branch 'trunk' into pcp-370-onboarding

AlexP11223 committed:
Remove progressive onboarding state

AlexP11223 committed:
Change onboarding buttons text

AlexP11223 committed:
Refactor settings element state requirements checking

AlexP11223 committed:
Allow to switch between live and sandbox

AlexP11223 committed:
Fix missing fields when switching from disconnected to onboarded env

AlexP11223 committed:
Validate manual input

AlexP11223 committed:
Fix ui

AlexP11223 committed:
Move JS from global scope

Dinamiko committed:
Merge branch 'pcp-370-onboarding' of github.com:woocommerce/woocommerce-paypal-payments into pcp-370-onboarding

Dinamiko committed:
Merge branch 'trunk' into pcp-370-onboarding

Dinamiko commented on modules/ppcp-wc-gateway/src/Settings/SettingsRenderer.php:
```diff
@@ -12,6 +12,7 @@
 use WooCommerce\PayPalCommerce\AdminNotices\Entity\Message;
 use WooCommerce\PayPalCommerce\ApiClient\Helper\DccApplies;
 use WooCommerce\PayPalCommerce\Button\Helper\MessagesApply;
+use WooCommerce\PayPalCommerce\Onboarding\Environment;
```
Seems `Environment` is not used in this class.

Dinamiko commented on modules/ppcp-onboarding/src/State.php:
```diff
@@ -109,37 +115,37 @@ public function production_state() : int {
+	}
+
 	/**
 	 * Returns the state based on progressive and onboarded values being looked up in the settings.
```
Maybe update description (... _based on progressive_ ...) here as progressive is not used anymore.

Dinamiko reviewed (approved):
Stunning work as usual, I´ve just added a couple of little comments.

AlexP11223 committed:
Remove unused

AlexP11223 committed:
Fix sandbox checkbox

Sometimes the state of UI may get restored after page reload and will not match the server settings

AlexP11223 committed:
Update phpdoc

AlexP11223 committed:
Remove unused import

Dinamiko committed:
Fix merge conflicts

Dinamiko committed:
Merge branch 'pcp-370-onboarding' of github.com:woocommerce/woocommerce-paypal-payments into pcp-370-onboarding
